### ISSUE TITLE
Problem with Backup replication

### ISSUE THREAD

**Issue by chino23** [pouchdb/pouchdb#5909]
I have set up a backup replication to Cloudant so the user can backup his data.

Now the problem is, when I restore it with a reverse replication, it doesn't really push anything back.
Trying it in incognito mode where the previous pouchdb is empty it works just fine.

That's probably as he's checking the time stamps of changes and realizes that it's not the newest elements and therefore doesn't update them, am I right?

Is there a way to implement that kind of backup system? Do I have to destroy() the database first, or what's the best approach here?

Thanks!

**Comment by nolanlawson** [comment]
Sorry, can you explain in a bit more detail what’s going on here? PouchDB doesn’t replicate based on timestamps; it replicates based on sequence numbers and replications are idempotent, meaning that if replication is not working, then it’s either a very serious bug or some kind of HTTP proxy intercepting the traffic and doing bad things with it. Can you provide more detail and potentially steps to reproduce? Do you see any errors in the Network panel?

**Comment by chino23** [comment]
Hi, thanks for the reply.

Ok, so here's some more info.

1) First I create a bunch of documents.

2) Then I call my replication script to make a backup to Cloudant
````
PouchDB.replicate(pouchdb.database, 'https://'+username+':'+password+'@xxx.cloudant.com/' + pouchdb.databaseName)
.on('complete', function (info) {
                        console.log('complete');
                        console.log(info);
                        $rootScope.$emit('cloud:success');
                    })
````
The info log shows all is well:
````
Object
doc_write_failures : 0
docs_read : 4
docs_written : 4
end_time : Tue Nov 22 2016 08:06:42 GMT+0800 (PHT)
errors : Array[0]
last_seq : 9
ok : true
start_time : Tue Nov 22 2016 08:06:29 GMT+0800 (PHT)
status : "complete"
__proto__ : Object
````
This works just fine, and checking online in the cloudant dashboard it's all there, as it should be. So far, so good.

3) I then change up a few docs on the app, delete some, etc. I reach a point where I say "damn, I have to roll back to that previous backup)

4) I now want to restore the old backup, and run the reverse replication script
````
PouchDB.replicate('https://'+username+':'+password+'@xxx.cloudant.com/' + pouchdb.databaseName, pouchdb.database)
.on('complete', function (info) {
                    console.log('complete');
                    console.log(info);
                    $rootScope.$emit('restore:success');
                })
````
the output of info is
````
doc_write_failures : 0
docs_read : 4
docs_written : 4
end_time : Tue Nov 22 2016 08:13:28 GMT+0800 (PHT)
errors : Array[0]
last_seq : "46-g1AAAAGXeJzLYWBgYMlgTmGQS0lKzi9KdUhJMjTSK87JzEvXS87JL01JzCvRy0styQGqY0pkSLL___9_VgZzokQuUIA9MdnEwjgxlaBukAWycCUWuMxPcgCSSfVQKzjAVpglJqYaG5kS0o_mBWNcVuSxAEmGBiAFtGU_yBpmiE_MLM2TkhMJGkKkTyDWHIBYA_YNN9iaJItEw6Rkgr7JAgBHB4Mp"
ok : true
start_time : Tue Nov 22 2016 08:13:20 GMT+0800 (PHT)
status : "complete"
__proto__ : Object
````

Seems like it all worked, BUT, the data in my app is still exactly the same as before I tried to restore the backup.

So, Replication -> Cloudant works fine, all there.
Replication back from Cloudant shows it's working, but the data on the pouch never changes back even though the console.info shows it's all ok.


----
Now when I create a incognito Chrome window, login, then there's obviously no data in pouch as it's a new session. Then I hit the restore from Cloudant button, and it works, it's ALL there, just as it should be.

So to recap, backup works fine.
Restore from Backup only seems to work in a fresh incognito window


So what I'm thinking is happening here:
When I try to restore, the documents in the Cloudant backup have lower revisions than the ones locally. Deleting a document just means I actually add
````
_deleted: true
````
So the doc is technically still in the db ,with doc_id and _rev, but the rev is higher.

Then replicating back it doesn't put any documents back because they are all older revision, is this correct?

That would explain why it works just flawlessly in the incognito session, as there are no newer _revs available.

**Comment by willholley** [comment]
@chino23 I think you're misunderstanding how replication works - it is sync'ing the revision history for each document, not the latest version. If a revision that is backed up to Cloudant is updated on the local datastore, replication of that back from Cloudant would be a no-op because the revision already exists locally. The [CouchDB documentation](http://docs.couchdb.org/en/2.0.0/replication/conflicts.html#couchdb-replication) has a good section on this.

The short of it is that you'll need to delete/recreate your local database before restoring from the backup.

**Comment by chino23** [comment]
yeah you're right.

I've solved it now by destroying the database first, then replicating the data back, which works just fine. Thanks!
